scripts/mcp-monitor.sh

```
@@ -0,0 +1,357 @@
+log_error() {
+    echo -e "${RED}[ERROR]${NC} $1"
+    echo "[$(date '+%Y-%m-%d %H:%M:%S')] ERROR: $1" >> "$LOG_FILE"
+}
+
+log_metric() {
+    echo "[$(date '+%Y-%m-%d %H:%M:%S')] $1" >> "$METRICS_FILE"
+}
+
+# Function to check file descriptor usage
+check_fd_usage() {
+    local current_fd=$(ulimit -n)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)